turbolinks/src/main/java/com/basecamp/turbolinks/TurbolinksJavascriptInjector.java

```diff
@@ -0,0 +1,74 @@
+package com.basecamp.turbolinks;
+
+import android.content.Context;
+import android.util.Base64;
+import android.webkit.ValueCallback;
+import android.webkit.WebView;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class TurbolinksJavascriptInjector {
+    private static String scriptInjectionFormat = "(function(){var parent = document.getElementsByTagName('head').item(0);var script = document.createElement('script');script.type = 'text/javascript';script.innerHTML = window.atob('%s');parent.appendChild(script);return true;})()";
+
+    // ---------------------------------------------------
+    // Public
+    // ---------------------------------------------------
+
+    /**
+     * <p>Injects Javascript into the webView.</p>
+     *
+     * @param turbolinksSession The TurbolinksSession.
+     * @param context           Any Android context.
+     * @param webView           The shared webView.
+     * @param filePath          Local file path relative to the main/src directory.
+     */
+    public static void injectJavascript(final TurbolinksSession turbolinksSession, Context context, WebView webView, String filePath) {
+        try {
+            String script = TurbolinksJavascriptInjector.getContentFromAssetFile(context, filePath);
+            String jsCall = String.format(scriptInjectionFormat, script);
+
+            webView.evaluateJavascript(jsCall, new ValueCallback<String>() {
+                @Override
+                public void onReceiveValue(String s) {
+                    if (turbolinksSession != null) {
+                        turbolinksSession.turbolinksBridgeInjected = Boolean.parseBoolean(s);
```

Comment:
Ah, good catch, I missed the `turbolinksBridgeInjected` flag. I'll remove that when anything other than the Turbolinks bridge is injected.

And good point regarding the callback. I'll add that in as well.